packages/jupyterlab-chat-extension/src/chat-commands/providers/user-mention.tsx

```diff
@@ -126,8 +126,18 @@ class MentionCommandProvider implements IChatCommandProvider {
    */
   _getUsers(inputModel: IInputModel): Map<string, Private.CommandUser> {
     const users = new Map();
+    // chatContext should be of type `LabChatContext`, so `users` should be of
+    // type `LabChatUser[]`, where `mention_name` is always defined.
     const userList = inputModel.chatContext.users;
     userList.forEach(user => {
+      if (!user.mention_name) {
+        console.error(
+          `No 'mention_name' property for user '${user.username}'. ` +
+            'This user is being omitted from ' +
+            "'MentionCommandProvider._getUsers()'."
+        );
+        return;
+      }
       users.set(user.mention_name, {
```

Comment:
```suggestion
    const userList = inputModel.chatContext.users as LabChatUser[];
    userList.forEach(user => {
      users.set(user.mention_name, {
```

Since the `chatContext` is a `LabChatContext`, the user list should always be a `LabChatUser[]` type.

Reply:
This is intentional even though I agree this is safe right now. The root issue is that `IChatContext.users` is not the same as `LabChatContext.users`, so it's possible that `LabChatContext.users` returns a user without a `mention_name`. In this case, we should log an error instead of ignoring the user silently.